app/api/study-plan/route.ts

```diff
@@ -59,9 +50,72 @@ export async function GET(req: NextRequest) {
         },
       };
     });
+};
+
+export async function GET(req: NextRequest) {
+  const studyPlanCollaborator = await getCollaborator(req);
+
+  if (!studyPlanCollaborator || studyPlanCollaborator.length === 0) {
+    return NextResponse.json({}, { status: 401 });
+  }
+
+  // TODO: Another todo, we are always taking the studyplanner with index 0, possibly needs be some kind of use preference or last viewed or something
```

Comment:
this will come from a url param, the route will be changed to /api/study-plans/:studyPlanId. the client will know the studyPlanId either from the url or from GET /api/study-plans. let's already change the endpoint urls (also in the client), ignore `studyPlanId` in the backend for now, and send /api/study-plans/foo or something from the client